thriftbp/server_middlewares.go

```diff
@@ -380,3 +383,59 @@ func RecoverPanic(name string, next thrift.TProcessorFunction) thrift.TProcessor
 		},
 	}
 }
+
+// PrometheusMetrics tracks Prometheus metrics specific to the Thrift service.
+func PrometheusMetrics(service string) thrift.ProcessorMiddleware {
+	return func(method string, next thrift.TProcessorFunction) thrift.TProcessorFunction {
+		process := func(ctx context.Context, seqID int32, in, out thrift.TProtocol) (success bool, err thrift.TException) {
+			start := time.Now()
+			var successLabel, exceptionTypeLabel, baseplateStatusCode, baseplateStatus string
+			activeRequestLabels := prometheus.Labels{
+				"thrift_service":               service,
+				"thrift_method":                method,
+				"thrift_success":               successLabel,
+				"thrift_exception_type":        exceptionTypeLabel,
+				"thrift_baseplate_status":      baseplateStatus,
+				"thrift_baseplate_status_code": baseplateStatusCode,
+			}
+			activeRequests.With(activeRequestLabels).Inc()
```

Comment:
do we really need all those labels for active requests? isn't it that when we do the `Inc` all those labels are empty strings anyways?

Reply:
I believe in the current state `Dec` will use the correct labels, because we saved the whole labels map and they are strings not pointers, but yes this is very error-prone and could easily go wrong in the future.